### DISCUSSION TITLE
Where is the mediatr handler for ExecuteWorkflowInstanceRequest

### DISCUSSION THREAD

**Discussion by shykln** [elsa-workflows/elsa-core#3375] [Q&A]
Hi, 

I have a question about how mediatr is used here:

https://github.com/elsa-workflows/elsa-core/blob/0eb9014e16fd5a46549b84a993f63b9e1147c121/src/server/Elsa.Server.Orleans/Grains/WorkflowInstanceGrain.cs#L16
```public async Task ExecuteWorkflowAsync(ExecuteWorkflowInstanceRequest request, CancellationToken cancellationToken = default) => await _mediator.Send(request, cancellationToken);```

I could not find where ```ExecuteWorkflowInstanceRequest``` is handled by mediatr. Could you kindly help point out? 
Thank you!

**Comment by sfmskywalker** [comment]
The `WorkflowInstanceGrain` isn't fully implemented, and I think that although it's sending the request to a mediatr, that there's no actual handler.

Instead, when using the `QueuingWorkflowDispatcher`, it also uses `ExecuteWorkflowInstanceRequest` by sending an instance of it to the service bus, for which there is a message handler: https://github.com/elsa-workflows/elsa-core/blob/master/src/core/Elsa.Core/Services/Dispatch/Consumers/ExecuteWorkflowInstanceRequestConsumer.cs

As far as I could find, this is the only place this message is handled.
